src/project/internal/projectconfiguration.cpp

```diff
@@ -263,14 +265,19 @@ async::Channel<io::path_t> ProjectConfiguration::userProjectsPathChanged() const
     return m_userScoresPathChanged;
 }
 
-io::path_t ProjectConfiguration::cloudProjectsPath() const
+io::path_t ProjectConfiguration::cloudProjectPath(const io::path_t& projectName) const
 {
-    return globalConfiguration()->userAppDataPath() + "/cloud_projects";
+    return cloudProjectsPath() + '/' + projectName + '.' + engraving::MSCZ;
```

Comment:
Are we sure that `projectName` contains no scary characters like `'/'`? I was personally considering the option to use the score id that's also used on MuseScore.com.

Reply:
> I was personally considering the option to use the score id

I don't like the idea to use scoreId because it becomes available at a pretty late point in time. A ton of things can go wrong while uploading the score to musescore.com. It is much safer to save the project before sending it so that if anything happens the user does not lose their data. But you are absolutely right that using a filename also has its drawbacks. Let's keep things as they are for now, I'll think about this more closely a bit later

Reply:
Yes, the fact that we don't know the id yet for new cloud scores is one of the problems for which I couldn't think of a solution yet. 
Of course, we can simply encode the filename by replacing special characters. We have `io::escapeFileName`, but I'm not sure how "waterproof" this method is. Another issue with this method is that if we don't watch out, the user will see the escaped version of the name in "Recent Scores", which is not ideal.
Maybe this is a comical idea, but alternatively we can base64 encode the name and use the result as the filename, and decode it before showing it in Recent Scores. QByteArray has builtin methods for this. 

A disadvantage of basing the filename on the score name is that it might case issues when the user renames the score on MuseScore.com. Of course, we need to store the source URL / score id at least _somewhere_, so we can still find out that the score has been renamed and that we'll need to rename it in our `cloud_projects` folder too, but it might be slightly more complicated. 

Anyway, I agree, let's keep it as it is for now, and give it some thought.